kafka-client/src/main/java/dev/responsive/kafka/internal/db/CassandraWindowedTable.java

```diff
@@ -99,37 +100,8 @@ public class CassandraWindowedTable implements
   private final PreparedStatement reserveEpoch;
   private final PreparedStatement ensureEpoch;
 
-  // Note: we intentionally track stream-time separately here and in the state store itself
-  // as these entities have different views of the current time and should not be unified.
-  // (Specifically, this table will always lag the view of stream-time that is shared by the
-  // ResponsiveWindowStore and CommitBuffer due to buffering/batching of writes)
-  private final Map<Integer, PendingFlushInfo> kafkaPartitionToPendingFlushInfo = new HashMap<>();
-
-  // TODO: move this into the LWTWriter/Factory to keep this class stateless
-  private static class PendingFlushInfo {
-
-    private long lastFlushStreamTime;
-    private long pendingFlushStreamTime;
-    private SegmentRoll segmentRoll;
-
-    public PendingFlushInfo(final long persistedStreamTime) {
-      this.lastFlushStreamTime = persistedStreamTime;
-      this.pendingFlushStreamTime = persistedStreamTime;
-    }
-
-    void maybeUpdatePendingStreamTime(final long recordTimestamp) {
-      this.pendingFlushStreamTime = Math.max(pendingFlushStreamTime, recordTimestamp);
-    }
-
-    void initSegmentRoll(final SegmentRoll pendingSegmentRoll) {
-      this.segmentRoll = pendingSegmentRoll;
-    }
-
-    void finalizeFlush() {
-      segmentRoll = null;
-      lastFlushStreamTime = pendingFlushStreamTime;
-    }
-  }
+  private final ConcurrentMap<Integer, SegmentBatch> kafkaPartitionToSegmentMetadata =
```

Comment:
ideally this class is "stateless" in the sense that it doesn't carry information per-partition. what do you think about moving this as well as all the precommit/postcommit code into the FlushManager (now that we have one per table per partition), which is already per partition IIUC? (basically you don't have a map here)

the benefit is that we don't recycle tables during a rebalance, so it's nice to know that if a rebalance happens twice and we're reassigned a partition we were assigned a long time ago we just create a new thing and never accidentally use corrupted old data from an old assignment